src/ServiceControl/Operations/ErrorIngestionComponent.cs

```diff
@@ -44,28 +42,37 @@ class ErrorIngestionComponent
             ErrorIngestionCustomCheck.State ingestionState
         )
         {
-            receivedMeter = metrics.GetCounter("Error ingestion - received");
+            var receivedMeter = metrics.GetCounter("Error ingestion - received");
             batchSizeMeter = metrics.GetMeter("Error ingestion - batch size");
             var ingestedMeter = metrics.GetCounter("Error ingestion - ingested");
             var bulkInsertDurationMeter = metrics.GetMeter("Error ingestion - bulk insert duration", FrequencyInMilliseconds);
             batchDurationMeter = metrics.GetMeter("Error ingestion - batch processing duration", FrequencyInMilliseconds);
 
             this.settings = settings;
-            var announcer = new FailedMessageAnnouncer(domainEvents);
-            persister = new ErrorPersister(documentStore, bodyStorageEnricher, enrichers, failedMessageEnrichers, ingestedMeter, bulkInsertDurationMeter);
-            ingestor = new ErrorIngestor(persister, announcer, settings.ForwardErrorMessages, settings.ErrorLogQueue);
+            var errorProcessor = new ErrorProcessor(bodyStorageEnricher, enrichers, failedMessageEnrichers, domainEvents, ingestedMeter);
+            var retryConfirmationProcessor = new RetryConfirmationProcessor(domainEvents);
 
-            var ingestion = new ErrorIngestion(async messageContext =>
-                {
-                    var taskCompletionSource = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
-                    messageContext.SetTaskCompletionSource(taskCompletionSource);
+            ingestor = new ErrorIngestor(errorProcessor, retryConfirmationProcessor, documentStore, bulkInsertDurationMeter, settings.ForwardErrorMessages, settings.ErrorLogQueue);
+
+            async Task Process(MessageContext messageContext)
```

Comment:
maybe move to the end of the function to be less interrupting when reading through the method

Reply:
The message filtering logic has been refactored and there is no longer a local function defined.